### PR TITLE
Move cloud tests to new repo

### PR DESCRIPTION
**Pull request by delucchi-cmu**
Adds `cloud_tests` from hipscat and lsdb repos.

- unifies the `conftest` set up of various cloud environment variables
- future PRs will
  - add tests for hipscat-import
  - benchmarks that use data hosted in the cloud bucket

### CODE REVIEW

File: benchmarks/benchmarks.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+
+def time_computation():
+    """Time computations are prefixed with 'time'."""
+    example_benchmarks.runtime_computation()
+
+
+def mem_list():
+    """Memory computations are prefixed with 'mem' or 'peakmem'."""
+    return example_benchmarks.memory_computation()
```
Comment (smcguire-cmu) [inline benchmarks/benchmarks.py:16]: Do we want these example benchmarks?

File: src/hipscat_cloudtests/example_module.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+def meaning() -> int:
+    """The meaning of life, the universe, and everything.
+
+    Returns
+    -------
+    int
+        The meaning of life.
+    """
+    return 42
```
Comment (smcguire-cmu) [inline src/hipscat_cloudtests/example_module.py:23]: Do we need this file?

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+def pytest_addoption(parser):
+    parser.addoption("--cloud", action="store", default="abfs")
+
+
+def pytest_generate_tests(metafunc):
+    # This is called for every test. Only get/set command line arguments
+    # if the argument is specified in the list of test "fixturenames".
+    option_value = metafunc.config.option.cloud
+    if "cloud" in metafunc.fixturenames and option_value is not None:
+        metafunc.parametrize("cloud", [option_value])
```
Comment (smcguire-cmu) [inline tests/conftest.py:20]: this is cool!

File: tests/hipscat/io/test_write_metadata_cloud.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+        "total_rows": 10,
+        "epoch": "J2000",
+        "ra_column": "ra",
+        "dec_column": "dec",
+    }
+
+
+@pytest.fixture
+def catalog_info(catalog_info_data) -> CatalogInfo:
+    return CatalogInfo(**catalog_info_data)
```
Comment (smcguire-cmu) [inline tests/hipscat/io/test_write_metadata_cloud.py:45]: Could these be moved to a `conftest.py` file so they can be reused later if needed?

File: tests/lsdb/conftest.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+
+@pytest.fixture
+def test_data_dir_cloud(example_cloud_path):
+    return os.path.join(example_cloud_path, "lsdb", "data")
+
+
+@pytest.fixture
+def small_sky_dir_cloud(test_data_dir_cloud):
+    return os.path.join(test_data_dir_cloud, SMALL_SKY_DIR_NAME)
```
Comment (smcguire-cmu) [inline tests/lsdb/conftest.py:23]: I see a lot of these are duplicated between the hipscat and lsdb tests, do you think it's good to keep them seperate over putting them in a higher level shared contest?

File: src/hipscat_cloudtests/example_module.py (hunk 2 above)
Comment (delucchi-cmu) [reply src/hipscat_cloudtests/example_module.py:23]: Just for now. Created https://github.com/astronomy-commons/hipscat-cloudtests/issues/2 to force m'self to add some benchmarks, but this file keeps things compiling.

File: tests/lsdb/conftest.py (hunk 5 above)
Comment (delucchi-cmu) [reply tests/lsdb/conftest.py:23]: The full paths are different, since the `test_data_dir_cloud` is different between lsdb and hipscat.

**Review by smcguire-cmu — APPROVED** [review]

**Pull request merged**